### PR TITLE
Initial implementation of a command history

### PR DESCRIPTION
**Pull request by Sadraukar**
Initial implementation of a command history that can be recalled with the UP and DOWN arrow keys.

### CODE REVIEW

File: static/index.html [hunk 1]
```diff
@@ -8,6 +8,6 @@
 	<body onload="return mud.initPage();">
 		<textarea id="output" readonly="true"></textarea>
 		<br />
-		<input type="text" id="entry" onkeypress="return mud.onEntryKeyPress(this, event);" />
+		<input type="text" id="entry" onkeydown="return mud.onEntryKeyPress(this, event);" />
```
Comment (Sadraukar) [inline static/index.html:11]: I needed to change this because "onkeypress" does not cover the arrow keys.

File: static/scripts.js [hunk 2]
```diff
@@ -39,40 +41,93 @@ var mud = {}; // Namespace
...
+	};
+
+	mud.isDownKeyPress = function (oEvent) {
+		var keynum = mud.captureKeyNum(oEvent);
+
+		// Detect DOWN key
+		return (40 == keynum);
+	};
+
+	mud.captureKeyNum = function (oEvent) {
```
Comment (Sadraukar) [inline static/scripts.js:119]: Encapsulated this into it's own function rather than having it repeat in three places.

**Comment by Sadraukar** [comment]
One side-effect of this currently is that hitting the DOWN arrow key in most browsers causes the cursor in the text box to jump to the end of the input (this is just default behavior of hitting down on a single line text input).  If this is undesirable, it can be fixed.

**Comment by MartinSnyder** [comment]
This seems to work great on IE and Chrome, but does not work in Firefox.

Also, I would suggest a change: when you press up, save the current input into the history (even/especially if it is a blank line). That way if someone hits up accidentally, they can get back to where they started from. I just confirmed that this is how bash works.

**Pull request merged**